jersey/src/main/java/com/netflix/conductor/server/resources/TaskResource.java

```diff
@@ -32,192 +32,282 @@
-	public String requeue(@PathParam("taskType") String taskType) throws Exception {
-		return "" + taskService.requeuePendingTasks(taskType);
-	}
+    private ExecutionService taskService;
+    private boolean auth_referer_bypass;
+    private Configuration config;
+    private QueueDAO queues;
+    private WorkflowExecutor executor;
+
+    @Inject
+    public TaskResource(ExecutionService taskService, QueueDAO queues, WorkflowExecutor executor, Configuration config) {
+        this.taskService = taskService;
+        this.auth_referer_bypass = Boolean.parseBoolean(config.getProperty("workflow.auth.referer.bypass", "false"));
+        this.config = config;
+        this.queues = queues;
+        this.executor = executor;
+    }
+
+    @GET
+    @Path("/poll/{tasktype}")
+    @ApiOperation("Poll for a task of a certain type")
+    @Consumes({MediaType.WILDCARD})
+    public Task poll(@PathParam("tasktype") String taskType, @QueryParam("workerid") String workerId, @QueryParam("domain") String domain) throws Exception {
+        List<Task> tasks = taskService.poll(taskType, workerId, domain, 1, 100);
+        if (tasks.isEmpty()) {
+            return null;
+        }
+        return tasks.get(0);
+    }
+
+    @GET
+    @Path("/poll/batch/{tasktype}")
+    @ApiOperation("batch Poll for a task of a certain type")
+    @Consumes({MediaType.WILDCARD})
+    public List<Task> batchPoll(
+            @PathParam("tasktype") String taskType,
+            @QueryParam("workerid") String workerId,
+            @QueryParam("domain") String domain,
+            @DefaultValue("1") @QueryParam("count") Integer count,
+            @DefaultValue("100") @QueryParam("timeout") Integer timeout
+
+    ) throws Exception {
+        return taskService.poll(taskType, workerId, domain, count, timeout);
+    }
+
+    @GET
+    @Path("/in_progress/{tasktype}")
+    @ApiOperation("Get in progress tasks.  The results are paginated.")
+    @Consumes({MediaType.WILDCARD})
+    public List<Task> getTasks(@PathParam("tasktype") String taskType, @QueryParam("startKey") String startKey,
+                               @QueryParam("count") @DefaultValue("100") Integer count) throws Exception {
+        return taskService.getTasks(taskType, startKey, count);
+    }
+
+    @GET
+    @Path("/in_progress/{workflowId}/{taskRefName}")
+    @ApiOperation("Get in progress task for a given workflow id.")
+    @Consumes({MediaType.WILDCARD})
+    public Task getPendingTaskForWorkflow(@PathParam("workflowId") String workflowId, @PathParam("taskRefName") String taskReferenceName)
+            throws Exception {
+        return taskService.getPendingTaskForWorkflow(taskReferenceName, workflowId);
+    }
+
+    @POST
+    @ApiOperation("Update a task")
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "Authorization", dataType = "string", paramType = "header")})
+    public String updateTask(TaskResult task, @Context HttpHeaders headers) throws Exception {
```

Comment:
This might not work as WC uses task update. Please check WC in order to add Auth support if that possible at all (since we are using Netflix library)